linera-chain/src/data_types.rs

```diff
@@ -180,6 +180,84 @@ impl IncomingBundle {
             (message_id, posted_message)
         })
     }
+
+    /// Rearrange the messages in the bundle so that the first message is an `OpenChain` message.
+    /// Returns whether the `OpenChain` message was found at all.
+    pub fn put_openchain_at_front(bundles: &mut [IncomingBundle]) -> bool {
+        let Some(index) = bundles.iter().position(|msg| {
+            matches!(
+                msg.bundle.messages.first(),
+                Some(PostedMessage {
+                    message: Message::System(SystemMessage::OpenChain(_)),
+                    ..
+                })
+            )
+        }) else {
+            return false;
+        };
+
+        Self::put_front_and_shift(bundles, index);
```

Comment:
I think you're just reimplementing this:

```rust
bundles[0..=index].rotate_right(1);
```

Reply:
I replaced the content of `put_fron_and_shift` with 
```rust
v[0..=index].rotate_right(1);
```
and tests fail. Do you want me to investigate why or can I leave the current implementation?

Reply:
I'd investigate why: `rotate_right` seems to be meant to do exactly what you implemented below?

Reply:
OK, it failed w/ `out of index` error for empty slices. You were right - it's equivalent.

Reply:
You're right. I wanted to keep the wrapper method for unit testing but if we use `rotate_right` then it should be tested in std.